src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AccessModifier.java

```diff
@@ -0,0 +1,63 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.naming;
+
+import java.util.Locale;
+
+/**
+ * This enum represents access modifiers.
+ * Access modifiers names are taken from JLS:
+ * https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.6
+ *
+ * @author Andrei Selkin
+ */
+public enum AccessModifier {
+    /** Public access modifier. */
+    PUBLIC,
+    /** Protected access modifier. */
+    PROTECTED,
+    /** Package-local access modifier. */
+    PACKAGE_LOCAL,
```

Comment:
From where you get "package local" term ?

I see only "package" and "package-private":
>or package-private (no explicit modifier).

at https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html

> If a class or interface type is declared with package access, then it may be accessed only from within the package in which it is declared. 

at https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html#jls-6.6.1

>  In the absence of an access modifier, most declarations have package access. A class or interface type declared without an access modifier implicitly has package access. 

at https://docs.oracle.com/javase/specs/jls/se8/html/jls-6.html
http://bugs.java.com/view_bug.do?bug_id=4456057 (Java representative in evaluation also call it "package" )

---------------

I would rather stay with JDK terminology - "package access".
"package-private" looks like is more human friendly term, I do not think we should use it.
@rnveach , please share your opinion.